### ISSUE TITLE
[UX] Direct deposit: Figure out what codes for API changes for PPIU mean from a front-end user perspective

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#4003]
## Background

TL; DR — We need to know what these indicator codes mean and how they are used to know how they may or may not affect the end user experience for direct deposit.

On 12/3/19, our team was informed that EVSS has made changes to PPIU services to expose the missing indicator data that EVSS itself relies on for determining if Direct Deposit changes can be made for use by VA.gov. This will go out in the 1/8 eBenefits release. 

As part of this email, we received this documentation on the code:

[Update PPIU JSON with new element.docx](https://github.com/department-of-veterans-affairs/va.gov-team/files/3933643/Update.PPIU.JSON.with.new.element.docx)

On 12/5/19, we had a meeting with the EVSS team about this work. After this meeting, they sent us along the following indicators that we need to figure out how to translate for the frontend user experience:

PPIU API   control information | PPIU conversion from BGS | BGS   EBenefitsAddressUpdateWebService | BGS definition
-- | -- | -- | --
canUpdateAddress | True if all other indicators are true | N/A |  
corpAvailIndicator | True if corpAvailInd is not ‘N’ | corpAvailInd | corpAvailInd – the database is   available, service was able to reach Corporate
corpRecFoundIndicator | True if corpRecFoundInd is not ‘N’ | corpRecFoundInd | corpRecFoundInd – a PERSON record was   found for the information passed in
indentityIndicator | True if indentityInd is not ‘N’ | indentityInd | indentityInd – The identity traits   passed in match what is stored on Corporate
indexIndicator | True if indexInd is not ‘B’ | indexInd | indexInd – The record’s Index value,   if there is one
isCompetentIndicator | True if incompetentInd is not ‘Y’ | incompetentInd | incompetentInd – whether the Veteran   is incompetent or not
mailingAddressExistIndicator | True if mailAddrExstInd is ‘Y’ | mailAddrExstInd | mailAddrExstInd – did an address   exist of type ‘Mailing’
noFiduciaryAssignedIndicator | True if fiduciaryInd is not ‘Y’ | fiduciaryInd | fiduciaryInd – whether the record has   a Fiduciary or not
notDeceasedIndicator | True if deathDtInd is not ‘Y’ | deathDtInd | deathDtInd – a DEATH_DT was found on   the PERSON record
paymentAddressExistIndicator | True if pymntAddrExstInd is ‘Y’ | pymntAddrExstInd | pymntAddrExstInd – did an address   exist of type ‘CP Payment’
  |   |   |  
PPIU API control information |   | BGS PaymentInformationService |  
hasNoBDNPaymentsIndicator | True if user has never received a payment type = ”Benefits for Children   of Vietnam Veterans (inc. Spina Bifida)” | paymentType |  

## Tasks

In order to move forward, we need to figure out what the above codes mean from a user perspective.

- [x] Set up meeting with eBenefits/EVSS where we get them to tell us what these codes mean from a front-end user perspective
- [ ] Break down design and development work into tickets

## Acceptance Criteria:

- We understand what the new API codes mean well enough to create design/development tickets

**Comment by Samara-Strauss** [comment]
Sent a follow up email on 12/18/19 to the EVSS folks to see if anyone could tell us who to talk to about what these codes mean.

**Comment by Samara-Strauss** [comment]
Followed up again on 1/29/20.

**Comment by Samara-Strauss** [comment]
Latest update: BGS doesn't know what these indicators mean from a business/user perspective. They recommended we circle back with Melissa Rebstock to see if she can help.

**Comment by Samara-Strauss** [comment]
BGS sent us along some definitions of what these indicators mean from their perspective. I still haven't had any luck getting eBenefits to tell me what any of these mean from a user/UX perspective.

They also sent definitions along for these indicators, but these were not included in the original spreadsheet above:


- **awardSuspBadAddrsInd** – is the Award suspended do to a bad   address
- **awardSuspInd** –   is the award   suspended
- **invalidRoutingInd** – did the update contain an invalid routing   number
- **updateConfirmation** – update success of fail
- **emailAddrExstInd** – did an address exist of type ‘Email’
- **phoneNumExstInd** – whether a phone number was found

**Comment by Samara-Strauss** [comment]
Additional clarification from an email with BGS on 2/6/2020:

> **indentityInd** – The identity traits passed in match what is stored on Corporate
>  
> What identity traits is this talking about?
>  
> Sorry I’m not sure which method is being used. One of them accepts lastName, ssn, and dob. Those would need to match what is stored on Corporate. This was how eBenefits wanted it years ago, before web services they called our EJB locally.  
>  
> **indexInd** – The record’s Index value, if there is one
>  
> What does “record” mean in this case? What is an “index value”?
>  
> The record would be the person associated to the attributes passed in, or the ptcpnt_id passed in. Index is an old legacy concept that told which system a person was getting paid out of, BDN or Corporate.

**Comment by Samara-Strauss** [comment]
Update: We now know what to do with the competency/fiduciary/deceased indicators — hide the DD feature from people who have these.

**Comment by Samara-Strauss** [comment]
I actually think we have what we need here. Need to regroup with the team and figure out how we can move forward.

**Comment by Samara-Strauss** [comment]
I don't think there is any additional work that needs to happen here. Should have closed this months ago :)
